engine/gamesys/src/gamesys/components/comp_collision_object.cpp

```diff
@@ -638,13 +668,44 @@ namespace dmGameSystem
             dmGameObject::HInstance instance_b = component_b->m_Instance;
             dmhash_t instance_a_id = dmGameObject::GetIdentifier(instance_a);
             dmhash_t instance_b_id = dmGameObject::GetIdentifier(instance_b);
-
-            dmPhysicsDDF::ContactPointResponse ddf;
             float mass_a = dmMath::Select(-contact_point.m_MassA, 0.0f, contact_point.m_MassA);
             float mass_b = dmMath::Select(-contact_point.m_MassB, 0.0f, contact_point.m_MassB);
+            uint64_t group_hash_a = GetLSBGroupHash(world, contact_point.m_GroupA);
+            uint64_t group_hash_b = GetLSBGroupHash(world, contact_point.m_GroupB);
 
-            uint64_t group_hash_a = GetLSBGroupHash(cud->m_World, contact_point.m_GroupA);
-            uint64_t group_hash_b = GetLSBGroupHash(cud->m_World, contact_point.m_GroupB);
+            if (world->m_LuaListener != 0)
+            {
+                dmPhysicsDDF::ContactPoint a;
+                a.m_Group               = group_hash_a;
+                a.m_Id                  = instance_a_id;
+                a.m_Position            = dmGameObject::GetWorldPosition(instance_a);
+                a.m_Mass                = mass_a;
+                a.m_RelativeVelocity    = -contact_point.m_RelativeVelocity;
+                a.m_Normal              = -contact_point.m_Normal;
+
+                dmPhysicsDDF::ContactPoint b;
+                b.m_Group               = group_hash_b;
+                b.m_Id                  = instance_b_id;
+                b.m_Position            = dmGameObject::GetWorldPosition(instance_b);
+                b.m_Mass                = mass_b;
+                b.m_RelativeVelocity    = contact_point.m_RelativeVelocity;
+                b.m_Normal              = contact_point.m_Normal;
+
+                dmPhysicsDDF::ContactPointEvent ddf;
+                ddf.m_A = a;
+                ddf.m_B = b;
+                ddf.m_AppliedImpulse = contact_point.m_AppliedImpulse;
+                ddf.m_Distance = contact_point.m_Distance;
+
+                dmGameObject::Result message_result = dmGameObject::PostDDF(&ddf, 0x0, &world->m_LuaListenerReceiver, world->m_LuaListener, false);
+                if (message_result != dmGameObject::RESULT_OK)
+                {
+                    dmLogError("Error when sending ContactPointEvent : %d", message_result);
+                }
+                return true;
+            }
+
+            dmPhysicsDDF::ContactPointResponse ddf;
```

Comment:
It would be great if we could deprecate the old ways later on as well...

Reply:
yes! at some point we will!